node/pkg/watchers/solana/ccq.go

```diff
@@ -182,6 +193,109 @@ func (w *SolanaWatcher) ccqHandleSolanaAccountQueryRequest(ctx context.Context,
 	w.ccqSendQueryResponse(queryRequest, query.QuerySuccess, resp)
 }
 
+// ccqCheckForMinSlotContext checks to see if the returned error was due to the min context slot not being reached. If so, and the estimated time in the future is not too great, it kicks off
+// a go routine to sleep and do a retry. In that case, it returns true, telling the caller that it is handling the request so it should not post a response. Note that the go routine only does
+// a single retry, but may result in another go routine being initiated to do another, and so on.
+func (w *SolanaWatcher) ccqCheckForMinSlotContext(ctx context.Context, queryRequest *query.PerChainQueryInternal, req *query.SolanaAccountQueryRequest, requestId string, err error, giveUpTime time.Time, log bool) bool {
+	if req.MinContextSlot == 0 {
+		return false
+	}
+
+	if time.Now().After(giveUpTime) {
+		w.ccqLogger.Info("giving up on fast retry", zap.String("requestId", requestId))
+		return false
+	}
+
+	isMinContext, currentSlot, err := ccqIsMinContextSlotError(err)
+	if err != nil {
+		w.ccqLogger.Error("failed to parse for min context slot error", zap.Error(err))
+		return false
+	}
+
+	if !isMinContext {
+		return false
+	}
+
+	// Estimate how far in the future the requested slot is, assuming a slot time of 400 ms.
+	msInTheFuture := (req.MinContextSlot - currentSlot) * 400
+
+	// If the requested slot is more than ten seconds in the future, use the regular retry mechanism.
+	if msInTheFuture > 10000 {
+		w.ccqLogger.Info("minimum context slot is too far in the future, requesting slow retry",
+			zap.String("requestId", requestId),
+			zap.Uint64("currentSlot", currentSlot),
+			zap.Uint64("minContextSlot", req.MinContextSlot),
+			zap.Uint64("msInTheFuture", msInTheFuture),
+		)
+		return false
+	}
+
+	// Kick off the retry after a short delay.
+	go w.ccqSleepAndRetryAccountQuery(ctx, queryRequest, req, requestId, currentSlot, giveUpTime, log)
+	return true
+}
+
+const CCQ_FAST_RETRY_INTERVAL = 200
+
+// ccqSleepAndRetryAccountQuery does a short sleep and then initiates a retry.
+func (w *SolanaWatcher) ccqSleepAndRetryAccountQuery(ctx context.Context, queryRequest *query.PerChainQueryInternal, req *query.SolanaAccountQueryRequest, requestId string, currentSlot uint64, giveUpTime time.Time, log bool) {
+	if log {
+		w.ccqLogger.Info("minimum context slot has not been reached, will retry shortly",
+			zap.String("requestId", requestId),
+			zap.Uint64("currentSlot", currentSlot),
+			zap.Uint64("minContextSlot", req.MinContextSlot),
+			zap.Int("retryInterval", CCQ_FAST_RETRY_INTERVAL),
+		)
+	}
+
+	time.Sleep(CCQ_FAST_RETRY_INTERVAL * time.Millisecond)
+
+	if log {
```

Comment:
same, seems like this is worthwhile to log even on the first retry, but I could be missing something

Reply:
Same response as above.